loadtest/setup_server.go

```diff
@@ -197,79 +197,80 @@ func SetupServer(cfg *LoadTestConfig) (*ServerSetupData, error) {
+		*serverConfig.TeamSettings.EnableOpenServer = true
+		if _, resp := adminClient.UpdateConfig(serverConfig); resp.Error != nil {
+			mlog.Error("Failed to set EnableOpenServer", mlog.Err(resp.Error))
+			return resp.Error
 		}
+	}
 
-		mlog.Info("EnableOpenServer is true")
+	mlog.Info("EnableOpenServer is true")
 
-		if *serverConfig.TeamSettings.MaxUsersPerTeam < 50000 {
-			mlog.Info("Attempting to set MaxUsersPerTeam to 50000 for the load test.", mlog.Int("old_max_users_per_team", *serverConfig.TeamSettings.MaxUsersPerTeam))
-			*serverConfig.TeamSettings.MaxUsersPerTeam = 50000
-			if _, resp := adminClient.UpdateConfig(serverConfig); resp.Error != nil {
-				mlog.Error("Failed to set MaxUsersPerTeam", mlog.Err(resp.Error))
-				return resp.Error
-			}
+	if *serverConfig.TeamSettings.MaxUsersPerTeam < 50000 {
+		mlog.Info("Attempting to set MaxUsersPerTeam to 50000 for the load test.", mlog.Int("old_max_users_per_team", *serverConfig.TeamSettings.MaxUsersPerTeam))
+		*serverConfig.TeamSettings.MaxUsersPerTeam = 50000
+		if _, resp := adminClient.UpdateConfig(serverConfig); resp.Error != nil {
+			mlog.Error("Failed to set MaxUsersPerTeam", mlog.Err(resp.Error))
+			return resp.Error
 		}
+	}
 
-		mlog.Info(fmt.Sprintf("MaxUsersPerTeam is %v", *serverConfig.TeamSettings.MaxUsersPerTeam))
+	mlog.Info(fmt.Sprintf("MaxUsersPerTeam is %v", *serverConfig.TeamSettings.MaxUsersPerTeam))
 
-		if *serverConfig.TeamSettings.MaxChannelsPerTeam < 50000 {
-			mlog.Info("Attempting to set MaxChannelsPerTeam to 50000 for the load test.", mlog.Int64("old_max_channels_per_team", *serverConfig.TeamSettings.MaxChannelsPerTeam))
-			*serverConfig.TeamSettings.MaxChannelsPerTeam = 50000
-			if _, resp := adminClient.UpdateConfig(serverConfig); resp.Error != nil {
-				mlog.Error("Failed to set MaxChannelsPerTeam", mlog.Err(resp.Error))
-				return resp.Error
-			}
+	if *serverConfig.TeamSettings.MaxChannelsPerTeam < 50000 {
+		mlog.Info("Attempting to set MaxChannelsPerTeam to 50000 for the load test.", mlog.Int64("old_max_channels_per_team", *serverConfig.TeamSettings.MaxChannelsPerTeam))
+		*serverConfig.TeamSettings.MaxChannelsPerTeam = 50000
+		if _, resp := adminClient.UpdateConfig(serverConfig); resp.Error != nil {
+			mlog.Error("Failed to set MaxChannelsPerTeam", mlog.Err(resp.Error))
+			return resp.Error
 		}
+	}
 
-		mlog.Info(fmt.Sprintf("MaxChannelsPerTeam is %v", *serverConfig.TeamSettings.MaxChannelsPerTeam))
+	mlog.Info(fmt.Sprintf("MaxChannelsPerTeam is %v", *serverConfig.TeamSettings.MaxChannelsPerTeam))
 
-		if !serverConfig.ServiceSettings.EnableIncomingWebhooks {
-			mlog.Info("Enabing incoming webhooks for the load test...")
-			serverConfig.ServiceSettings.EnableIncomingWebhooks = true
-			if _, resp := adminClient.UpdateConfig(serverConfig); resp.Error != nil {
-				mlog.Error("Failed to set EnableIncomingWebhooks", mlog.Err(resp.Error))
-				return resp.Error
-			}
+	if !*serverConfig.ServiceSettings.EnableIncomingWebhooks {
```

Comment:
These are pointers now. The server also seems to complain about "missing" new configuration options, which was failing the loadtest @cpanato was running.

Reply:
👍